backend/src/api/board-history-get.ts

```diff
@@ -13,7 +13,29 @@ export const boardHistoryGet = route
     .use(apiTokenHeader)
     .handler((request) =>
         checkBoardAPIAccess(request, async (board) => {
-            const history = await withDBClient((client) => getFullBoardHistory(board.board.id, client))
-            return ok({ history })
+            return ok(
+                streamingBody(async (stream) => {
+                    // Due to memory concerns we fetch board histories from DB as chunks, so this API
+                    // response must also be chunked
+                    stream.write('{"history":[')
```

Comment:
This is cool, but it might be even cooler to intruduce a little abstraction here - separating the "spit strings to stream" from the "business logic" part.